### PR TITLE
Implemented HTML post-processor to attach external images as mail attachments, add missing host to link urls

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Also added a feature to refresh templates from source, and a mail specific template content.

### CODE REVIEW

File: mail_editor/settings.py [hunk 1]
```diff
@@ -57,6 +57,15 @@ def DYNAMIC_CONTEXT(self):
...
             return import_string(dynamic)
 
+    @property
+    def BASE_HOST(self):
+        """
+        protocol, domain and (optional) port, no ending slash
+
+        used to retrieve images for embedding and fix URLs
+        """
+        return getattr(django_settings, 'MAIL_EDITOR_BASE_HOST', "")
```
Comment (Bartvaderkin) [inline mail_editor/settings.py:67]: Not sure if we want a separate setting for this or pull it from somewhere else

File: mail_editor/models.py [hunk 2]
```diff
@@ -144,8 +147,7 @@ def render(self, context, subj_context=None):
         body_context.update({'content': partial_body})
         body = template_function(self.base_template_path, body_context)
 
-        # TODO: I made the package.json stuff optional. Maybe it should be removed completely since it adds 3 settings,
-        # seems for a limited use-case, and it uses subproces...
```
Comment (Bartvaderkin) [inline mail_editor/models.py:148]: Jiro tested this and it works, so we'll keep it.

File: mail_editor/process.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+
+def process_html(html, base_url="", extract_attachments=True, fix_links=True):
+    parser = etree.HTMLParser()
+    root = etree.fromstring(html, parser)
+
+    image_attachments = dict()
+    url_cid_cache = dict()
+
+    if extract_attachments:
```
Comment (sergei-maertens) [inline mail_editor/process.py:40]: I'd recommend using mail-cleaner as dependency for this instead: https://github.com/maykinmedia/mail-cleaner/blob/main/mail_cleaner/mail.py#L92

It's your code from Open Forms in a separate package :)

File: mail_editor/settings.py (hunk 1 above)
Comment (sergei-maertens) [reply mail_editor/settings.py:67]: easy enough to do `MAIL_EDITOR_BASE_HOST = BASE_URL` in a project's settings, namespacing is good so I'd stick with this :)

File: mail_editor/process.py (hunk 3 above)
Comment (Bartvaderkin) [reply mail_editor/process.py:40]: It is not quite the same; the linked code does data-urls, while this does http url's.

The intent is render standard Django templates with static and url tags and everything.

Anyway, I did isolate it a little bit so we can lift it to mail-cleaner later.

File: mail_editor/process.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+
+def load_image(url):
+    import requests
+    # TODO check domains?
+    # TODO check data urls
+    r = requests.get(url)
+    content_type = r.headers["Content-Type"]
+    if not content_type.startswith("image/"):
+        raise Exception("not an image file")
```
[the remark this answers is not in the document]
Comment (alextreme) [reply mail_editor/process.py:13]: Why not use the Attachment+ImageField approach with FK to the MailTemplate? Then you can use whatever Django storage backend you want

File: mail_editor/models.py (hunk 2 above)
Comment (Bartvaderkin) [reply mail_editor/models.py:148]: Swapped it with CSS inliner

**Comment by alextreme** [comment]
Doen we in de feature/mailpreview branch, komt nog een PR van feature/mailpreview naar master

**Pull request merged**